src/Dfe.Spi.EventBroker.Functions/Subscriptions/UpdateSubscription.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfe.Spi.Common.Http.Server;
+using Dfe.Spi.Common.Logging.Definitions;
+using Dfe.Spi.Common.Models;
+using Dfe.Spi.EventBroker.Application.Subscriptions;
+using Dfe.Spi.EventBroker.Domain.Subscriptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Dfe.Spi.EventBroker.Functions.Subscriptions
+{
+    public class UpdateSubscription : FunctionsBase<UpdateSubscriptionRequest>
+    {
+        private readonly ISubscriptionManager _subscriptionManager;
+        private readonly ILoggerWrapper _logger;
+
+        public UpdateSubscription(
+            ISubscriptionManager subscriptionManager,
+            ILoggerWrapper logger)
+            : base(logger)
+        {
+            _subscriptionManager = subscriptionManager;
+            _logger = logger;
+        }
+
+        [FunctionName("UpdateSubscription")]
+        public async Task<IActionResult> RunAsync(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "subscriptions")]
+            HttpRequest req,
+            CancellationToken cancellationToken)
+        {
+            return await ValidateAndRunAsync(req, cancellationToken);
+        }
+
+        protected override HttpErrorBodyResult GetMalformedErrorResponse()
+        {
+            return new HttpErrorBodyResult(
+                HttpStatusCode.BadRequest,
+                "NOTJSON",
```

Comment:
I've noticed that the various `HttpErrorBody` result codes are in different formats. Just a small thing, but would be nice to tidy up. I'll point out the other places you create these instances.